src/script/check_commands.sh

```diff
@@ -1,19 +1,22 @@
 #!/bin/sh
-git grep COMMAND\( | grep -o "(\"[a-zA-z ]*\"" | grep -o "[a-zA-z ]*" > commands.txt
+
+git grep COMMAND\( | awk -F'"' 'NF && !seen[$2]++ {} END { a=asorti(seen,alf); for(i=1;i<=a;i++) print alf[i]}' > commands.txt
+# Another way would be to pipe the output to: grep -Po '(?<=\(")[^"]+' | sort | uniq  but this will unnecessarily fork sort and uniq
 missing_test=false
-good_tests=""
-bad_tests=""
+good_tests=()
+bad_tests=()
 while read cmd; do
     if git grep -q "$cmd" -- src/test qa/; then
-	good_tests="$good_tests '$cmd'"
+	good_tests+=("$cmd")
     else
 	echo "'$cmd' has no apparent tests"
 	missing_test=true
-	bad_tests="$bad_tests '$cmd'"
+	bad_tests+=("$cmd")
     fi
 done < commands.txt
 
-if [ "$missing_test" == true ]; then
-    echo "Missing tests!" $bad_tests
+if $missing_test; then
+    printf '\n%s\n' "Missing tests"
```

Comment:
why `printf` is considered better than `echo`?

Reply:
@tchaikov In this particular script echo will do but printf is also portable and has a rich fomatting options for different data types. I used printf since I intended to use arrays. Futhermore, there is a lot wrong with echo. See for example https://wiki.bash-hackers.org/commands/builtin/echo or https://www.in-ulm.de/~mascheck/various/echo+printf/